src/app/dashboard/page.js

```
@@ -1,15 +1,125 @@
+        label: 'Donations (MAD)',
+        data: donationData,
+        borderColor: '#42A5F5',
+        backgroundColor: 'rgba(66, 165, 245, 0.2)',
+        tension: 0.4,
+      },
+    ],
+  };
+
+  // for protected route 
+  const router = useRouter();
+  useEffect(() => {
```

ESLint

[error] React Hook "useEffect" is called conditionally. React Hooks must be called in the exact same order in every component render.

(react-hooks/rules-of-hooks)